core/src/main/scala/laserdisc/protocol/ClusterP.scala

```diff
@@ -32,62 +36,69 @@ object ClusterP {
     private final val A: String ==> Address = {
       val HPCP = raw"([A-Za-z0-9\-\.]*):(\d+)@(\d+)".r
       val HP   = raw"([A-Za-z0-9\-\.]*):(\d+)".r
-      Read.instancePF {
-        case HPCP("", ToInt(Port(p)), ToInt(Port(cp)))      => Address(LoopbackHost, p, cp)
-        case HPCP(Host(h), ToInt(Port(p)), ToInt(Port(cp))) => Address(h, p, cp)
-        case HP("", ToInt(Port(p)))                         => Address(LoopbackHost, p, p)
-        case HP(Host(h), ToInt(Port(p)))                    => Address(h, p, p)
+      Read.instance {
+        case HPCP("", ToInt(Port(p)), ToInt(Port(cp)))      => Right(Address(LoopbackHost, p, cp))
+        case HPCP(Host(h), ToInt(Port(p)), ToInt(Port(cp))) => Right(Address(h, p, cp))
+        case HP("", ToInt(Port(p)))                         => Right(Address(LoopbackHost, p, p))
+        case HP(Host(h), ToInt(Port(p)))                    => Right(Address(h, p, p))
+        case other                                          => Left(Err(s"Unexpected encoding for Address. Was $other"))
       }
     }
-    private final val Fs: String ==> Seq[Flag] = Read.instancePF {
-      case "noflags" => Seq.empty
+    private final val Fs: String ==> Seq[Flag] = Read.instance {
+      case "noflags" => Right(Seq.empty)
       case s =>
-        s.split(COMMA_CH).toList.collect {
-          case "myself"    => Flag.myself
-          case "master"    => Flag.master
-          case "slave"     => Flag.replica
-          case "fail?"     => Flag.possiblefail
-          case "fail"      => Flag.fail
-          case "handshake" => Flag.handshake
-          case "noaddr"    => Flag.noaddress
+        s.split(COMMA_CH).foldRight[Err | List[Flag]](Right(Nil)) {
+          case ("myself", Right(fgs))    => Right(Flag.myself :: fgs)
+          case ("master", Right(fgs))    => Right(Flag.master :: fgs)
+          case ("slave", Right(fgs))     => Right(Flag.replica :: fgs)
+          case ("fail?", Right(fgs))     => Right(Flag.possiblefail :: fgs)
+          case ("fail", Right(fgs))      => Right(Flag.fail :: fgs)
+          case ("handshake", Right(fgs)) => Right(Flag.handshake :: fgs)
+          case ("noaddr", Right(fgs))    => Right(Flag.noaddress :: fgs)
+          case (other, Right(_))         => Left(Err(s"Unexpected flag encoding. Was $other"))
+          case (_, Left(e))              => Left(e)
```

Comment:
```suggestion
          case (_, left)                 => left
```
or it doesn't typecheck?